client/java/src/main/java/io/openlineage/client/transports/KafkaConfig.java

```diff
@@ -13,14 +13,18 @@
 @ToString
 public final class KafkaConfig implements TransportConfig {
   @Getter @Setter private String topicName;
-  @Getter @Setter private String localServerId;
+  @Getter @Setter private String messageKey;
   @Getter @Setter private Properties properties;
 
   KafkaConfig() {
     properties = new Properties();
   }
 
-  public boolean hasLocalServerId() {
-    return (localServerId != null);
+  String getLocalServerId() {
+    return messageKey;
+  }
+
+  void setLocalServerId(String localServerId) {
+    this.messageKey = localServerId;
```

Comment:
Those fields are read by Jackson from Yaml. Would `localServerId` still work after this change? 
Desired solution is to make `localServerId` as deprecated with some javadoc comment telling this will be removed 3 versions ahead.

Reply:
> Would localServerId still work after this change?

There is a test `testConfToKafkaConfigLegacyLocalServerId`

> Desired solution is to make localServerId as deprecated with some javadoc comment telling this will be removed 3 versions ahead.

Done